### ISSUE TITLE
Interval.direction() doesn't distinguish accidentals of same note

### ISSUE THREAD

**Issue by eagor** [saebekassebil/teoria#105]
```
        var a = teoria.note('a'),
            ab = teoria.note('ab'),
            aS = teoria.note('a#');
        ab.interval(a).direction(); // "up"
        // but
        a.interval(ab).direction(); // "up"
        aS.interval(a).direction(); // "up"
        aS.interval(ab).direction(); // "up"
```
interval from A# to Ab returns direction "up", even though Ab is 2 semitones down from A#.

**Comment by saebekassebil** [comment]
Hey Eagor.

I can see why that might be confusing behaviour. But actually it is working as "expected": The `Interval#direction()` method doesn't measure whether the interval goes up or down *on the keyboard* - but rather which direction the interval as *"facing"*.

I'm not sure that there is such an interval as "diminished unison", but I kind of had to make it up to make the theory "complete".

I'm very open to listen to other suggestions on this behaviour if you have any input :)

**Comment by eagor** [comment]
Yeah, it's a bit misleading, as initially I would think that if you play two notes one after another, you would hear whether the melody goes up or down, hence the direction of the interval.

Or, maybe there is some other theoretical meaning of interval direction? Or what might be the use case expecting current behavior?

ps. anyways, great lib. The idea with .coord is genius!!!

**Comment by eagor** [comment]
maybe, this should go to a different thread, but I've just found this:
```
teoria.scale("cx", 'wholetone').simple()
> ["cx", "dx", "ex", "fundefined", "gundefined", "aundefined"]
```

**Comment by saebekassebil** [comment]
Well spotted! This should produce some kind of an error. Please open a new issue on this though.
